packages/loader/container-loader/src/deltaConnection.ts

```diff
@@ -74,35 +60,19 @@ export class DeltaConnection
             version: connection.version,
         };
 
-        connection.on("nack", (documentId: string, message: INack[]) => {
-            // Mark nacked and also pause any outbound communication
-            this._nacked = true;
-            this.emit("nack", documentId, message);
-        });
-
-        connection.on("disconnect", (reason) => {
-            this.emit("disconnect", reason);
-            this.close();
-        });
-
         this.on("newListener", (event: string, listener: (...args: any[]) => void) => {
             // Register for the event on connection
             // A number of events that are pass-through.
             // Note that we delay subscribing to op / signal on purpose, as
             // that is used as a signal in DocumentDeltaConnection to know if anyone has subscribed
             // to these events, and thus stop accumulating ops / signals in early handlers.
             // See DocumentDeltaConnection.initialMessages() implementation for details.
-            if (this.forwardEvents.includes(event)) {
```

Comment:
Although I also don't like the event forwarding logic here, the forwardEvents check protects against passing through events that don't match the `IDocumentDeltaConnectionEvents` set.  Perhaps we should keep it until we get rid of the event forwarding?

Reply:
I'd not spend any time here, but rather remove all this code in next PR right after this one.
That said, it's interesting question, as we do not do these kids of checks anywhere else, so it begs the question - should we? including in DocumentDeltaConnection layer.
And can be we it more generically in TypedEventEmitter?

Opened https://github.com/microsoft/FluidFramework/issues/3930 to track it.

Reply:
It's specific to the newListener and event forwarding behavior, so most of our classes have no problem.  Most event emitters can know exactly what events they emit (because they are the ones doing the emitting).  But with the re-forwarding, it becomes possible for DeltaConnection to emit any event that the document deltaconnection emits, which it can't know and will vary based on the document deltaconnection implementation.  The forwardEvents check is the current mechanism by which the DeltaConnection can know again exactly what events it emits.

Reply:
There are two problems:
1. How do we make sure that caller can register only on events that this class supports. This is the problem of TypedEventEmitter. Code like it exists here (before my changes) should not exist, it should be solved centrally, without creating new sources of information (list of events). The problem shows up today as "error" missing on signature of IDocumentDeltaConnectionEvents
2. How do we ensure in cases like that (forwarder) that adapter layer's (DeltaConnection) event set is strict subset of class it wraps. TypedEventEmitter (used by IDocumentDeltaConnection implementations) would help enforce it at runtime, but there is no solution for compile time (likely similar to # 1 - we are talking mostly about runtime checks, due to having generic on() handler).

I do not see an easy solution for # 2, and there never was one here.
\# 1 is more generic and is strictly a problem of TypedEventEmitter - I'm against of every TypedEventEmitter consumer maintaining duplicate list of events it knows about.

Reply:
I'll merge this PR, as this point is very moot either way - I'll raise PR right away to delete all this code :)